docs/operations.md

```diff
@@ -0,0 +1,825 @@
+
+It requires one [processor](#processor) callback to perform some work that's related to the input record.
+
+Its output record is a copy of the input, which means anything the callback returns is discarded.
+
+!!! info "Common uses"
+    * Non-transforming, record-related logic
+
+**Python API**
+```python
+class GearsBuilder.foreach(f)
+```
+
+_Arguments_
+
+* _f_: the [processor](#processor) function callback that will be used on each record
+
+**Examples**
+```python
+# Increment a shard-local counter for each record processed
+GB() \
+  .foreach(lambda x: execute('INCR', hashtag())) \
+  .run()
+```
+
+## Filter
+The local **Filter** operation performs one-to-zero-or-one (1:(0|1)) filtering of records.
+
+It requires a [filterer](#filterer) function callback.
+
+An input record that yields a falsehood will be discarded and only truthful ones will be output.
+
+!!! info "Common uses"
+    * Filtering records
+
+**Python API**
+```python
+class GearsBuilder.filter(f)
+```
+
+_Arguments_
+
+* _f_: the filtering function callback
+
+**Examples**
+```python
+# Filter out records where the key's length is less than four characters
+GB() \
+  .filter(lambda x: len(x['key']) > 3) \
+  .run()
+```
+
+## Accumulate
+The local **Accumulate** operation performs many-to-one mapping (N:1) of records.
+
+It requires one [accumulator](#accumulator) callback.
+
+Once input records are exhausted its output is a single record consisting of the accumulator's value.
+
+!!! info "Common uses"
+    * Aggregating records
+
+**Python API**
+```python
+class GearsBuilder.accumulate(f)
+```
+
+_Arguments_
+
+* _f_: an [accumulator](#accumulator) function callback
+
+**Examples**
+```python
+# Sum the lengths of all records' keys
+GB() \
+  .accumulate(lambda a, r: (a if a else 0) + len(r['key'])) \
+  .run()
+```
+
+## LocalGroupBy
+The local **LocalGroupBy** operation performs many-to-less mapping (N:M) of records.
+
+The operation requires two callbacks: an [extractor](#extractor) a [reducer](#reducer).
+
+The output records consist of the grouping key and its respective accumulator's value.
+
+!!! info "Common uses"
+    * Grouping records by key
+
+**Python API**
+```python
+class GearsBuilder.localgroupby(e, r)
+```
+
+_Arguments_
+
+* _e_: a key [extractor](#extractor) function callback
+* _r_: a [reducer](#reducer) function callback
+
+**Examples**
+```python
+# Locally group and count records by the first byte in their key
+GB() \
+  .localgroupby( \
+    lambda x: x['key'][:1],
+    lambda k, a, r: (a if a else 0) + 1) \
+  .run()
+```
+
+## Limit
+The local **Limit** operation limits the number of records.
+
+It accepts two numeric arguments: a starting position in the input records "array" and a maximal number of output records.
+
+!!! info "Common uses"
+    * Returning the first results
+    * Batch paging on static data
+
+**Python API**
+```python
+class GearsBuilder.limit(start, length)
```

Comment:
If only one argument is given its treated as length and start is set to zero.

Reply:
Fixed via 079f280